src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -901,10 +902,34 @@ protected virtual void Dispose(bool disposing)
 
         public void Dispose()
         {
+            StopWorkerProcess();
             _disposing = true;
             Dispose(true);
         }
 
+        private void StopWorkerProcess()
+        {
+            bool capabilityEnabled = !string.IsNullOrEmpty(_workerCapabilities.GetCapabilityState(RpcWorkerConstants.HandlesWorkerTerminateMessage));
+            if (!capabilityEnabled)
+            {
+                return;
+            }
+
+            var workerTerminate = new WorkerTerminate()
+            {
+                GracePeriod = Duration.FromTimeSpan(TimeSpan.FromSeconds(WorkerConstants.WorkerTerminateGracePeriodInSeconds))
+            };
+
+            _workerChannelLogger.LogDebug($"Sending WorkerTerminate message with grace period {WorkerConstants.WorkerTerminateGracePeriodInSeconds} seconds.");
+
+            SendStreamingMessage(new StreamingMessage
+            {
+                WorkerTerminate = workerTerminate
+            });
+
+            WorkerProcess.WaitForProcessExit();
```

Comment:
I feel like we should pass in the wait time here so it's clear when reading the code that we are waiting for the same grace period that we are sending in the WorkerTerminate message. It feels like this method should be in control of the time period it chooses rather than whatever is inside WaitForProcessExit().

```suggestion
            WorkerProcess.WaitForProcessExit(WorkerConstants.WorkerTerminateGracePeriodInSeconds);
```

Reply:
I agree. Updated the code here. Thanks.